### ISSUE TITLE
Site Settings: Leaving the app after opening the detailed discussion settings crashes the app

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-Android#3639]
Steps to reproduce:
1. Select a site and open Settings
2. Under "Discussion" select "More"
3. Leave the app (e.g. return to the device home screen)
4. Return to the app
Result: The message "Unfortunately, WordPress has stopped." appears, and you are returned to your site's My Sites screen.

Tested on Nexus 9, Android 6.0.1, app beta v. 5.0-rc-1.

**Comment by tonyr59h** [comment]
Another one I'm not able to reproduce on my HTC One or Nexus 5X. Is there any way you can grab a log of the exception?

**Comment by rachelmcr** [comment]
Sure, here's the log of the exception:

```
01-29 12:52:16.002  7381  7381 E AndroidRuntime: FATAL EXCEPTION: main
01-29 12:52:16.002  7381  7381 E AndroidRuntime: Process: org.wordpress.android, PID: 7381
01-29 12:52:16.002  7381  7381 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.prefs.BlogPreferencesActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v7.widget.Toolbar$SavedState
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.ActivityThread.-wrap11(ActivityThread.java)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:148)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v7.widget.Toolbar$SavedState
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readParcelableCreator(Parcel.java:2432)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readParcelable(Parcel.java:2358)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readValue(Parcel.java:2264)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readSparseArrayInternal(Parcel.java:2675)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readSparseArray(Parcel.java:1967)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readValue(Parcel.java:2321)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Parcel.readArrayMapInternal(Parcel.java:2614)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.BaseBundle.unparcel(BaseBundle.java:221)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.os.Bundle.getSparseParcelableArray(Bundle.java:856)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2033)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.Dialog.onRestoreInstanceState(Dialog.java:461)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:182)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:247)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:1823)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:321)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:326)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:326)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.Preference.restoreHierarchyState(Preference.java:1805)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:196)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.Fragment.performActivityCreated(Fragment.java:2228)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:992)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1148)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1130)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:1953)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:152)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.Activity.performCreateCommon(Activity.java:6245)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:6253)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
01-29 12:52:16.002  7381  7381 E AndroidRuntime: 	... 9 more
```

Let me know if you need anything more!

**Comment by tonyr59h** [comment]
That should be enough information. Thanks @rachelmcr :smile:

**Comment by tonyr59h** [comment]
Just found out in order to reproduce (reliably) `Don't keep activities` should be toggled on.

**Comment by tonyr59h** [comment]
Addressed in #3711.
